sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -103,6 +107,11 @@ def fit(self, X, y):
         acc_prediction_time = 0.
         X, y = check_X_y(X, y, dtype=[X_DTYPE], force_all_finite=False)
         y = self._encode_y(y)
+        if sample_weight is not None:
+            sample_weight = column_or_1d(sample_weight)
```

Comment:
can we use `_check_sample_weight`? (see also https://github.com/scikit-learn/scikit-learn/pull/14870)